src/_components/AddAsset/AddAssetContainer.jsx

```diff
@@ -0,0 +1,179 @@
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { ToastMessage } from '../../_utils/ToastMessage';
+
+import AddAssetComponent from '../../components/AddAsset/AddAssetComponent';
+
+// actions
+import { loadCategories } from '../../_actions/category.actions';
+import { loadSubCategories } from '../../_actions/subcategory.actions';
+import { loadAssetTypes } from '../../_actions/assetTypes.actions';
+import { loadAssetMakes } from '../../_actions/assetMakes.actions';
+import { loadModelNumbers } from '../../_actions/modelNumbers.actions';
+import { createAsset } from '../../_actions/asset.actions';
+
+// utils
+import {
+  filterSubCategories,
+  filterAssetTypes,
+  filterAssetMakes,
+  filterModelNumbers
+} from  '../../_utils/filterDropdowns';
+
+import _ from 'lodash';
+
+class AddAssetContainer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      filteredSubCategories: [],
+      filteredAssetTypes: [],
+      filteredAssetMakes: [],
+      filteredModelNumbers: [],
+      modelNumber: 0,
+      serialNumber: '',
+      assetTag: '',
+      assets: this.props.assets
+    };
+  }
+
+  componentDidMount() {
+    if (_.isEmpty(this.props.categories)) {
+      this.props.loadCategories();
+    }
+    if (_.isEmpty(this.props.subcategories)) {
+      this.props.loadSubCategories();
+    }
+    if (_.isEmpty(this.props.assetTypes)) {
+      this.props.loadAssetTypes();
+    }
+    if (_.isEmpty(this.props.assetMakes)) {
+      this.props.loadAssetMakes();
+    }
+    if (_.isEmpty(this.props.modelNumbers)) {
+      this.props.loadModelNumbers();
+    }
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.assets.length !== prevState.assets.length) {
+      ToastMessage.success({ message: 'Asset Saved Successfully' });
+      return {
+        filteredSubCategories: [],
+        filteredAssetTypes: [],
+        filteredAssetMakes: [],
+        filteredModelNumbers: [],
+        modelNumber: 0,
+        assets: nextProps.assets
+      }
+    }
+    return null;
+  }
+
+  handleDropdownChanges = (event, data) => {
+    if(data.name === 'asset-category') {
+      this.setState({
+        filteredSubCategories:
+        filterSubCategories(this.props.subcategories, data.value)
```

Comment:
Why the line break? If it is because the line is too long, then you can write the code like so:
```
this.setState({
  filteredSubCategories: filterSubCategories(
    this.props.subcategories,
    data.value
  )
});
```

Alternatively, destructure the values and use them, thereby decreasing the line length. E.g.
```
const { name, value } = data;
const { subcategories, assetTypes, assetMakes, modelNumbers } = this.props;
....
....

this.setState({
  filteredSubCategories: filterSubCategories(subcategories, value)
});
```

Please apply the changes based on whichever option you choose to all the _if statements_ in `handleDropdownChanges`.